docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -22,110 +24,54 @@ command:
 
     openssl rand 512 | base64
 
-Internal SSL/TLS configuration
-------------------------------
+Internal TLS configuration
+--------------------------
 
-SSL/TLS is configured in the ``config.properties`` file.  The SSL/TLS on the
-worker and coordinator nodes are configured using the same set of properties.
-Every node in the cluster must be configured. Nodes that have not been
-configured, or are configured incorrectly, are not able to communicate with
-other nodes in the cluster.
+You can configure the coordinator and all workers to encrypt all communication
+with each other using TLS. Every node in the cluster must be configured. Nodes
+that have not been configured, or are configured incorrectly, are not able to
+communicate with other nodes in the cluster.
 
-To enable SSL/TLS for Trino internal communication, do the following:
+In typical deployments you want to enable :ref:`TLS directly on the coordinator
+<https-secure-directly>` for fully encrypted access to the cluster by client
+tools.
 
-1. Disable HTTP endpoint.
+Now you can enable TLS for internal communication with the following
+configuration identical on all cluster nodes.
 
-   .. code-block:: text
-
-       http-server.http.enabled=false
-
-   .. warning::
-
-       You can enable HTTPS, while leaving HTTP enabled. In most cases this is a
-       security hole. If you are certain you want to use this configuration, you
-       should consider using an firewall to limit access to the HTTP endpoint to
-       only those hosts that should be allowed to use it.
-
-2. Configure the cluster to communicate using the fully qualified domain name (fqdn)
-   of the cluster nodes. This can be done in either of the following ways:
+1. Configure a shared secret for internal communication as described in
+   the preceding section.
 
-   - If the DNS service is configured properly, we can just let the nodes
-     introduce themselves to the coordinator using the hostname taken from
-     the system configuration (``hostname --fqdn``)
+2. Enable automatic certificate creation and trust setup in
+   ``etc/config.properties``:
 
-     .. code-block:: text
+   .. code-block:: properties
 
-         node.internal-address-source=FQDN
+     internal-communication.https.required=true
 
-   - It is also possible to specify each node's fully-qualified hostname manually.
-     This is different for every host. Hosts should be in the same domain to
-     make it easy to create the correct SSL/TLS certificates.
-     e.g.: ``coordinator.example.com``, ``worker1.example.com``, ``worker2.example.com``.
+   Older versions of Trino required you to manually manage all the certificates
+   on the nodes. If you upgrade from this setup, you need to ensure that
```

Comment:
Maybe say “you must remove the configuration properties XXX as they are no longer needed”